### ISSUE TITLE
USWDS - Bug: Step Indicator heading very difficult to understand with VoiceOver

### ISSUE THREAD

**Issue by jaredcunha** [uswds/uswds#6523]
### Describe the bug

The heading portion of the step indicator is nearly impossible to understand using VoiceOver.

### Steps to reproduce the bug

1. Turn on VoiceOver using MacOS
2. Using the virtual cursor, navigate to the heading portion of the step indicator

### Expected Behavior

The heading should should be announced in a way that is easily understood with any screen reader.

### Related code

It's like due to the number of child elements in the heading:

```html
<div class="usa-step-indicator__header">
    <h4 class="usa-step-indicator__heading">
      <span class="usa-step-indicator__heading-counter"
        ><span class="usa-sr-only">Step</span>
        <span class="usa-step-indicator__current-step">3</span>
        <span class="usa-step-indicator__total-steps">of 5</span> </span
      ><span class="usa-step-indicator__heading-text"
        >Supporting documents</span
      >
    </h4>
  </div>
```

A backwards-compatible solution would providing a section of screen reader text that contains a cleaner presentation of the text, while completely hiding the text that is visually presented.

```html
<h4 class="usa-step-indicator__heading">
   <!-- Screen reader only portion -->
   <span class="usa-sr-only">Step 1 of 5: Supporting documents</span>
   <!-- Visual presentation only portion --> 
   <span class="usa-step-indicator__heading-counter" aria-hidden="true"> 
      <span class="usa-step-indicator__current-step">1</span> 
      <span class="usa-step-indicator__total-steps">of <span>5</span> </span>
      <span class="usa-step-indicator__heading-text">Supporting documents</span> 
   </span>
</h4>
```

### Screenshots

The image below is what get announced
<img width="737" height="141" alt="Image" src="https://github.com/user-attachments/assets/7dc39557-6c71-4773-ad85-4ee094d40f0e" />

With the suggested fix, this is what we get (same code, different text)
<img width="1374" height="276" alt="Image" src="https://github.com/user-attachments/assets/dbe83820-2d85-4bdf-9de1-aa81fb70e329" />

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [x] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by josephine-njooi** [comment]
Chiming in that the text read out loud differs between browsers. Safari seems fine, but not Firefox or Chrome.

<details>
<summary>Safari</summary>

<img width="768" height="293" alt="Image" src="https://github.com/user-attachments/assets/dac25d5b-ad47-4e4c-9423-49d54978f9e6" />

</details>

<details>
<summary>Firefox</summary>

<img width="1343" height="705" alt="Image" src="https://github.com/user-attachments/assets/f2ead5bb-c74a-4fdd-a4b7-470489bfc331" />

</details>


<details>
<summary>Chrome</summary>

<img width="1522" height="948" alt="Image" src="https://github.com/user-attachments/assets/d200f023-dfb5-4345-b725-2202b7afb671" />


</details>


edit: I also don't know if I'm misremembering, but somehow I think there wasn't this issue before. I'm not sure if it's a change in VoiceOver/browser behavior.
